org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/secondgeneration/SecondGenerationLoginPost.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.binding.internal.kostal.inverter.secondgeneration;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.security.MessageDigest;
+import java.util.Base64;
+
+/**
+ * The {@link SecondGenerationLoginPost} is responsible for access to the inverter, regarded Post of changeable values,
+ * which are used in the second generation part of the binding.
+ *
+ * @author Örjan Backsell - Initial contribution Piko1020, Piko New Generation
+ */
+
+public class SecondGenerationLoginPost {
+    static final String USER_AGENT = "Mozilla/5.0";
+
+    // HTTP Sending Post Request
+    static int loginPost(String url, String username, String password, String salt) throws Exception {
+        URL objLoginPost = new URL(url);
+        HttpURLConnection conLoginPost = (HttpURLConnection) objLoginPost.openConnection();
```

Comment:
Please use openhab's provided HttpClient instead of java's native http connection services.